core-backend/sandbox/src/funcs.rs

```diff
@@ -401,6 +402,25 @@ where
         })
     }
 
+    pub fn random(ctx: &mut Runtime<E>, args: &[Value]) -> SyscallOutput {
+        sys_trace!(target: "syscall::gear", "random, args = {}", args_to_str(args));
+
+        let (random_ptr, subject_ptr, subject_len, bn_ptr): (u32, u32, u32, u32) =
```

Comment:
```suggestion
        let (random_ptr, subject_ptr, subject_len, bn_ptr): (u32, u32, u32, u32) =
```

arguments order messed up